remote_command_execution_vulnerability.py

```diff
@@ -131,7 +132,38 @@ def send_test_netspeed_request(router_ip_address, stok, port):
 else:  # Use remote github repository. port setted to 0.
     send_test_netspeed_request(router_ip_address, stok, port=0)
 
-print("done! Now you can connect to the router using several options: (user: root, password: root)")
-print("* telnet {}".format(router_ip_address))
-print("* ssh -oKexAlgorithms=+diffie-hellman-group1-sha1 -c 3des-cbc -o UserKnownHostsFile=/dev/null root@{}".format(router_ip_address))
-print("* ftp: using a program like cyberduck")
+
+retry = 3
+delay = 1
+timeout = 3
+def isOpen(ip, port):
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.settimeout(timeout)
+    try:
+        s.connect((ip, int(port)))
+        s.shutdown(socket.SHUT_RDWR)
+        return True
+    except:
+        return False
+    finally:
+        s.close()
+
+def checkHost(ip, port):
+    ipup = False
+    for i in range(retry):
+        if isOpen(ip, port):
+            ipup = True
+            break
+        else:
+            time.sleep(delay)
+    return ipup
+
+if checkHost(router_ip_address, 22):
+    print("done! Now you can connect to the router using several options: (user: root, password: root)")
+    print("* telnet {}".format(router_ip_address))
+    print("* ssh -oKexAlgorithms=+diffie-hellman-group1-sha1 -c 3des-cbc -o UserKnownHostsFile=/dev/null root@{}".format(router_ip_address))
+    print("* ftp: using a program like cyberduck")
+else:
+    print("Warning: the process has finished, but it seems that something goes wrong.")
+    print("* Maybe your firmware version is not supported, is it MiRouter 4A Gigabit 2.30.20 or 3.2.26?")

```

Comment:
Instead of `is it MiRouter 4A Gigabit 2.30.20 or 3.2.26?`, could you just say something like `please have a look at https://github.com/acecilia/OpenWRTInvasion/blob/master/README.md#unsupported-routers-and-firmware-versions`
So when the list of unsupported firmwares grow, we do not need to change it also here